src/Simplex_tree/test/simplex_tree_unit_test.cpp

```diff
@@ -991,3 +991,38 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(simplex_tree_reset_filtration, typeST, list_of_tes
 
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(simplex_tree_boundaries_and_opposite_vertex_iterator, typeST, list_of_tested_variants) {
+  std::clog << "********************************************************************" << std::endl;
+  std::clog << "TEST OF BOUNDARIES AND OPPOSITE VERTEX ITERATORS" << std::endl;
+  typeST st;
+
+  st.insert_simplex_and_subfaces({2, 1, 0}, 3.);
+  st.insert_simplex_and_subfaces({3, 0}, 2.);
+  st.insert_simplex_and_subfaces({3, 4, 5}, 3.);
+  st.insert_simplex_and_subfaces({0, 1, 6, 7}, 4.);
+
+  /* Inserted simplex:        */
+  /*    1   6                 */
+  /*    o---o                 */
+  /*   /X\7/                  */
+  /*  o---o---o---o           */
+  /*  2   0   3\X/4           */
+  /*            o             */
+  /*            5             */
+  using Simplex = std::vector<typename typeST::Vertex_handle>;
+  // simplices must be kept sorted by vertex number for std::vector to use operator== - cf. last BOOST_CHECK
+  std::vector<Simplex> simplices = {{0, 1, 2}, {0, 3}, {0, 1, 6, 7}, {3, 4, 5}, {3, 5}};
+  for (auto simplex : simplices) {
+    for(auto boundary_and_opposite_vertex : st.boundary_opposite_vertex_simplex_range(st.find(simplex))) {
+      Simplex output;
+      for (auto vertex : st.simplex_vertex_range(boundary_and_opposite_vertex.first)) {
+        std::clog << vertex << " ";
+        output.emplace_back(vertex);
+      }
+      std::clog << " - opposite vertex = " << boundary_and_opposite_vertex.second << std::endl;
+      output.emplace_back(boundary_and_opposite_vertex.second);
+      std::sort(output.begin(), output.end());
+      BOOST_CHECK(simplex == output);
```

Comment:
We check that what we output is indeed supposed to be part of the output. We don't test that we get all of them, and each one only once. I think it is ok not to test for redundancy, but maybe a simple counter checking that the size of the range is dim(simplex)+1 would be easy.

Reply:
Ok, I did a first try on [ff11b13](https://github.com/GUDHI/gudhi-devel/pull/597/commits/ff11b137365f0cef11062f9413907f0f39414b3e) but I was not happy with it. I prefer this version [7d1e88a](https://github.com/GUDHI/gudhi-devel/pull/597/commits/7d1e88a2995e23d882cc3bfe69ec09357d6c2618) of the test.